### PR TITLE
Factor out of my courses

### PR DESCRIPTION
**Pull request by cgalvind2l**

### CODE REVIEW

File: d2l-simple-overlay-styles.html [hunk 1]
```diff
@@ -0,0 +1,98 @@
+<link rel="import" href="../d2l-colors/d2l-colors.html">
```
Comment (cgalvind2l) [inline d2l-simple-overlay-styles.html]: This file is the same as it is in d2l-my-courses

File: d2l-simple-overlay.html [hunk 2]
```diff
@@ -0,0 +1,188 @@
+<link rel="import" href="../polymer/polymer.html">
```
Comment (cgalvind2l) [inline d2l-simple-overlay.html]: This file is not exactly the same as in d2l-my-courses, changes commented in this PR...

Comment (cgalvind2l) [inline d2l-simple-overlay.html]: First of all, `localize-behaviour` has been removed in favour of the langterm (only used on the close button) being passed to it as a parameter

File: d2l-simple-overlay.html [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+<dom-module id="d2l-simple-overlay">
+	<style include="d2l-simple-overlay-styles"></style>
+	<template>
+		<span role="dialog" aria-label$="[[titleName]]">
+			<div class="max-width container">
+				<h1 class="d2l-heading-2">{{titleName}}</h1>
+				<button
+					class="close-button"
+					on-tap="_handleClose"
+					aria-label$="[[closeSimpleOverlayAltText]]"
```
Comment (cgalvind2l) [inline d2l-simple-overlay.html]: `aria-label` and `alt` do not refer to localize

File: d2l-simple-overlay.html [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+				'neon-animation-finish': '_onNeonAnimationFinish',
+				'image-selector-tile-image-selected': '_handleCloseSimpleOverlayEvent'
+			},
+			properties: {
+				// Title for overlay
+				titleName: {
+					type: String
+				},
+				// Localized text for the close button label
+				closeSimpleOverlayAltText: {
```
Comment (cgalvind2l) [inline d2l-simple-overlay.html]: This property added to be the localized lang term for the close button.

Comment (jstefaniuk-d2l) [inline d2l-simple-overlay.html]: You can do `closeSimpleOverlayAltText: String` here

Comment (jstefaniuk-d2l) [inline d2l-simple-overlay.html]: And also maybe do it for titleName as well while you're here

File: d2l-simple-overlay-styles.html [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+			width: 100%;
+			border-top: 4px solid #003b71;
+			z-index: 1000;
+			user-select: none;
+			-webkit-user-select: none;
+			-moz-user-select: none;
+			-ms-user-select: none;
+		}
+
+		.close-button {
```
Comment (jstefaniuk-d2l) [inline d2l-simple-overlay-styles.html]: While we're here, do we want to implement the `d2l` prefixes?

File: d2l-simple-overlay-styles.html [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+			padding: 0 0 0.2rem 0;
+			color: var(--d2l-color-ferrite);
+			cursor: pointer;
+		}
+
+		.container {
+			display: flex;
+			justify-content: space-between;
+			align-items: center;
+			min-height: 105px;
```
Comment (jstefaniuk-d2l) [inline d2l-simple-overlay-styles.html]: This may be better as 5.25rem in case font sizes get shifted

File: d2l-simple-overlay-styles.html [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+			margin-left: auto;
+			margin-right: auto;
+			max-width: 1230px;
+			@apply(--simple-overlay-content-width);
+		}
+
+		h1 {
+			/* overrides to d2l-heading-2 (h1 styled with d2l-heading-2 class) */
+			font-weight: 400;
+			font-size: 30px;
```
Comment (jstefaniuk-d2l) [inline d2l-simple-overlay-styles.html]: I'd put 1.5rem here

File: d2l-simple-overlay-styles.html [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+		d2l-icon {
+			width: 12px;
+			height: 12px;
+		}
+
+		.scrollable {
+			height: calc(100vh - 100px);
+			max-height: calc(100vh - 100px);
+			overflow-y: scroll;
+			overflow-y: overlay;
```
Comment (jstefaniuk-d2l) [inline d2l-simple-overlay-styles.html]: Maybe @apalaniuk or @ryantmer can chip in as to why we're doing it this way, but I think `overlay: auto` may give less-strange behaviour on windows (the scroll bar overlays the content when we use overlay)

Comment (jstefaniuk-d2l) [inline d2l-simple-overlay-styles.html]: edited ^

**Comment by jstefaniuk-d2l** [comment]
Testing

File: d2l-simple-overlay-styles.html [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+		d2l-icon {
+			width: 12px;
+			height: 12px;
+		}
+
+		.scrollable {
+			height: calc(100vh - 100px);
+			max-height: calc(100vh - 100px);
+			overflow-y: scroll;
```
Comment (jstefaniuk-d2l) [inline d2l-simple-overlay-styles.html]: whoops, forgot to mention to remove `overflow-y: scroll`, it was just there previously as a fallback for `overflow-y: overlay`

**Comment by jstefaniuk-d2l** [comment]
Testing notes:
- Checked out all 3 repos and linked everything up locally
- Page loads
- Verified that the changes were loaded by checking the css
- Both the all-courses overlay and the image selector load
- I was able to close both overlays
- Text clears out of both overlays after closing

----- Tests failed ----
Problem:  Need to add `"postinstall": "bower install",` to scripts

- Ran `bower install` and then was able to run the tests

- Other than the one tiny issue, LGTM

**Comment by jstefaniuk-d2l** [comment]
LGTM

**Pull request merged**